File: A8-ToyLanguageInterpretatorGUI/src/Model/AssignStmt.java
package Model;

import Exceptions.DivisionByZero;
import Exceptions.OperatorNotFound;
import Exceptions.VariableNotFound;

public class AssignStmt implements IStmt {
    private String id;
    private Exp exp;

    public AssignStmt(String id, Exp exp){
        this.id = id;
        this.exp = exp;
    }

    public String toString() {
        return id + "=" + exp.toString();
    }


    public PrgState execute(PrgState state) throws DivisionByZero, VariableNotFound, OperatorNotFound {
        MyIDictionary<String, Integer> symTbl = state.getSymTable();
        MyIRandIntKeyDict<Integer> heap = state.getHeap();
        int val = exp.eval(symTbl,heap);
        if (symTbl.isDefined(id)) {
            symTbl.update(id, val);
        } else {
            symTbl.add(id, val);
        }
        return null;
    }
}

File: A7/ToyLanguage/Model/AssignStmt.cs
using System;
namespace ToyLanguage.Model
{
    public class AssignStmt : IStmt
    {
        private String id;
        private Exp exp;

        public AssignStmt(String id, Exp exp)
        {
            this.id = id;
            this.exp = exp;
        }

        public PrgState execute(PrgState state)
        {
            IDict<String, int> symTbl = state.getSymTbl();
            int val = exp.eval(symTbl);
            if (symTbl.isDefined(id))
            {
                symTbl.update(id, val);
            }
            else
            {
                symTbl.add(id, val);
            }
            return null;
        }
        public override string ToString()
        {
            return id + "=" + exp.ToString();
        }

    }
}
